src/engine/bufferscalers/enginebufferscalerubberband.cpp

```diff
@@ -142,31 +122,50 @@ void EngineBufferScaleRubberBand::clear() {
     VERIFY_OR_DEBUG_ASSERT(m_pRubberBand) {
         return;
     }
-    m_pRubberBand->reset();
+    reset();
 }
 
 SINT EngineBufferScaleRubberBand::retrieveAndDeinterleave(
         CSAMPLE* pBuffer,
         SINT frames) {
-    SINT frames_available = m_pRubberBand->available();
-    SINT frames_to_read = math_min(frames_available, frames);
+    const SINT frames_available = m_pRubberBand->available();
+    // NOTE: If we still need to throw away padding, then we can also
+    //       immediately read those frames in addition to the frames we actually
+    //       need for the output
+    const SINT frames_to_read = math_min(frames_available, frames + m_remainingPaddingInOutput);
+    DEBUG_ASSERT(frames_to_read <= m_buffers[0].size());
     SINT received_frames = static_cast<SINT>(m_pRubberBand->retrieve(
-            m_retrieve_buffer, frames_to_read));
+            m_bufferPtrs.data(), frames_to_read));
+    SINT frame_offset = 0;
+
+    // As explained below in `reset()`, the first time this is called we need to
+    // drop the silence we fed into the time stretcher as padding from the
+    // output
+    if (m_remainingPaddingInOutput > 0) {
+        const SINT drop_num_frames = std::min(received_frames, m_remainingPaddingInOutput);
+
+        m_remainingPaddingInOutput -= drop_num_frames;
+        received_frames -= drop_num_frames;
+        frame_offset += drop_num_frames;
+    }
 
+    DEBUG_ASSERT(received_frames <= frames);
     SampleUtil::interleaveBuffer(pBuffer,
-                                 m_retrieve_buffer[0],
-                                 m_retrieve_buffer[1],
-                                 received_frames);
+            m_buffers[0].data() + frame_offset,
+            m_buffers[1].data() + frame_offset,
+            received_frames);
+
     return received_frames;
 }
 
 void EngineBufferScaleRubberBand::deinterleaveAndProcess(
         const CSAMPLE* pBuffer, SINT frames, bool flush) {
+    DEBUG_ASSERT(frames <= static_cast<ssize_t>(m_buffers[0].size()));
```

Comment:
When I try to compile this locally on Windows with VS2022, it fails with an error, that identifier ssize_t is an undefined.

Reply:
Oh I didn't realize `ssize_t` isn't from the standard library, and it compiles fine with MSVC on the CI. Probably through some transitive dependency. I guess `ptrdiff_t` would be the portable alternative that comes closest to `ssize_t`.

Reply:
Afaik its part of the C standard library. I guess you could also `#include <cstddef>`?

Reply:
It doesn't appear to be. Libusb and the other libraries Mixxx uses that also use `ssize_t` seem to explicitly define it on Windows.